### ISSUE TITLE
All of my saved feeds vanished after update

### ISSUE THREAD

**Issue by mhork** [QuiteRSS/quiterss#1074]
I don't know why, none of my settings changed. I had about 30 feeds that all just disappeared after the update.

**Comment by davidmcl** [comment]
Mine as well ... over 40 feeds gone

OS: Linux Mint 18.3

**Comment by dbareis** [comment]
I didn't have this problem but thanks for reminding me to export my feeds
for a current backup.

**Comment by davidmcl** [comment]
Yes, you can backup - but you might notice there is no restore option :-) 

What appears to be the case is that they changed the location of the feeds.db table.

I located and copied the old table into the new location and got my feeds back - it would have been better for them to have copy/moved the old feeds.db to the new location and saved all the bother and confusion.

QuiteRss info was in the  `~/.local/share/data/QuiteRss/QuiteRss` directory ... they moved everything out from under the "...data" directory and put things in the `~/.local/share/QuiteRss/QuiteRss` directory which I can understand, but really should have been handled better than 'oops' - it's gone.

dbm

PS.  And if they wanted to be real nice they could add the db location as an specifiable option  like a couple other things that allow overrides, e.g., the style sheets

**Comment by Funcy-dcm** [comment]
We did not change anything. These are changes to Qt 5.x

**Comment by StargazerA5** [comment]
I just tried upgrading to 18.11 via the Ubuntu PPA and the result was a complete mess.  The feeds were present, but everything was marked unread and I lost all my labels.  The feed.db file had dropped from 66 MB to 30 MB.  I tried swaping the feed.db and QuiteRSS.ini with backup copies to no luck.  I finally did an uninstall, delete all traces of QuiteRSS in my Home directory, reinstall 18.11, and then restore the db and ini from backup to get everything working again.  However since the db and ini were made in an earlier version, I'm hoping I'm not introducing a compatibility issue somewhere.

**Comment by dbareis** [comment]
I haven't tried a backup. I actually just meant the actual feed information (.opml) which you can export and import.

I'm using a portable version so that could explain why I haven't had any issues.

**Comment by tonebacas** [comment]
I'm having problems with my feeds as well. I'm on Manjaro and installed QuiteRss from the repository for this linux distro (not AUR), and apparently QuiteRss has some problems with its database file.

I did some testing, and as it turns out, if there's no feeds.db file in `~/.local/share/QuiteRss/QuiteRss` (like a fresh install would seem), QuiteRss starts up and shows the default feeds, and then if you restart it, no feeds show up, and there's a newly created feeds.db file in that location. If you delete that file, QuiteRss will launch again with no problems and loads the default feeds again.

It seems it can't deal with the `feeds.db` file properly.

I'm running version 0.18.10 (16.04.2018).

**Comment by UserLambda2** [comment]
I'm a very basic end-user. Don't code. Don't know nothing about architecturing softwares and tied processes.
Just to say : I've been, but, very, very, *sad* to see my opml file deleted by the update process (0.18.10 to 0.18.11).
About 30 feeds, 15 stared articles (using QRSS since only 1 month) and 2 "Must read" gone with the wind.
Don't wanna blame anyone, since I really like the software, but maybe there's a way to warn dummies like me to back up before update.
Many thanks anyway for the great job.

**Comment by UserLambda2** [comment]
I have a ".bak" file (/home/userlambda2/.config/QuiteRss/backup), generated when I installed QRSS 1 month ago and imported an opml one. Not the paradise, but better than nothing. But QRSS won't deal with such file.

**Comment by tonebacas** [comment]
@UserLambda2 I assume you didn't manage to find your `feeds.db` file still intact? Maybe it's still there, it's just QuiteRss can't display it properly because of the behavior I described before, even if I'm using the older version instead of the new one.

**Comment by UserLambda2** [comment]
Thanks @tonebacas !
I didn't pay attention, but yes `feeds.db` is there. Unfortunately, it's no more intact, 'cause in a very pessimistic way, I started to modify it after posting here (added : 2 feeds on an blank sheet). 

But, let's say `feeds.db` was as before the update : what's the trick, then (if there's one) when update happens ? (Knowing that the graphic interface only proposes .opml and .xml choices).

[also knowing that, after the Update Lesson, I will regularly backup my config].

**Comment by tonebacas** [comment]
@UserLambda2 I assume QuiteRss would just keep using that `feeds.db` file, reading and writing your feeds like it was previously. That file should hold all your feeds and news, including your starred ones. This file expands as you fetch more news, and depending on the options you've set, it's cleaned frequently to get rid of old news. The `.opml` and `.xml` files are just the list of feeds you're subscribed to, not the news themselves, so it doesn't include your starred news.

If you got a hold of your `feeds.db` file, I assume you'd keep your old feeds and news as they were. Now, sadly it appears you've overwritten this file, so I don't have a solution for your problem.

**Comment by dbareis** [comment]
It would be nice if you could schedule a backup (and add a restore rather than having to manually copy files) or export of the database.  Currently you have to remember to do it, I've now got a recent backup but the last time I'd remembered to do it was in 2016 sometime.

**Comment by tonebacas** [comment]
@dbareis if you're on Linux use `cron` (https://linux.die.net/man/8/cron) and schedule your own `feeds.db` backups.
On Windows, use the task scheduler for the same effect.

One thing to note, though, if you're running QuiteRss while you do the backup, it may still be using the `feeds.db` file, so should close it before running a backup, or you might end up with an outdated representation of your feeds and news.

**Comment by dbareis** [comment]
Yep, thats basically why you can't schedule it now.  Who knows what you'll end up when the file is in use.  That is why it needs to be a schedule to be automated or I need to set up a batch file to start it.  Much easier and safer if the app backs up the database.

**Comment by host-noah0** [comment]
@davidmcl 
YES!!! This is the right way, I use the portable version, .bak file is placed in root directory of the program and rename it.
